sites/melbourne/lib/melbourne/data/llm_caller.rb

```diff
@@ -0,0 +1,167 @@
+# frozen_string_literal: true
+
+module Melbourne
+  module Data
+    class LLMCaller # rubocop:disable Metrics/ClassLength:
+      def initialize(openai_api_key:)
+        # Define the request URL
+        uri = URI('https://api.openai.com/v1/responses')
+
+        # Create HTTP client
+        @http = Net::HTTP.new(uri.host, uri.port)
+        @http.use_ssl = true
+
+        # Create the request
+        @request = Net::HTTP::Post.new(uri)
+        @request['Content-Type'] = 'application/json'
+        @request['Authorization'] = "Bearer #{openai_api_key}"
+      end
+
+      def call(path:)
+        meetups = JSON.parse(path)
+        meetups.map do |meetup|
+          perform_request(meetup)
+        end
+      end
+
+      private
+
+      attr_reader :request, :http
+
+      def perform_request(meetup)
+        request.body = request_body(meetup).to_json
+
+        response = http.request(request)
+        parsed_response = JSON.parse(response.body)
+
+        JSON.parse(parsed_response.fetch("output").first.fetch("content").first.fetch("text"))
+      end
+
+      def request_body(event_json) # rubocop:disable Metrics/MethodLength
```

Comment:
Same as above; these classes are just data processing scripts not meant to live forever or to be used regularly at all